src/Microsoft.AspNetCore.Mvc.Razor/RazorPageBase.cs

```diff
@@ -85,14 +85,18 @@ public virtual TextWriter Output
         /// <summary>
         /// Gets or sets a <see cref="System.Diagnostics.DiagnosticSource"/> instance used to instrument the page execution.
         /// </summary>
+#pragma warning disable PUB0001 // Pubternal type in public API
```

Comment:
I think checking attributes is too much. I'll disable this check.

Reply:
Yeah, attributes aren't a contract. We don't require user code to interact with them.